app/common/model/src/main/java/io/syndesis/common/model/integration/step/template/TemplateStepLanguage.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.common.model.integration.step.template;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Locale;
+import java.util.Map;
+import io.syndesis.common.model.Dependency;
+import io.syndesis.common.model.integration.Flow;
+import io.syndesis.common.model.integration.Integration;
+import io.syndesis.common.model.integration.Integration.Builder;
+import io.syndesis.common.model.integration.Step;
+import io.syndesis.common.model.integration.StepKind;
+import io.syndesis.common.util.StringConstants;
+
+/**
+ * Provides languages model for use with the template step feature.
+ * Each language has a name, maven dependency, camel header id
+ * and {@link TemplateStepPreProcessor} implementation defined
+ * for use by the template step handler and project generator.
+ */
+public class TemplateStepLanguage {
+
+    public static final String LANGUAGE_PROPERTY = "language";
+
+    public static final TemplateStepLanguage MUSTACHE = new TemplateStepLanguage(
+             "MUSTACHE",
+             "org.apache.camel:camel-mustache:2.21.0.fuse-720007",
```

Comment:
We should not have hard-coded version

Reply:
@lburgazzoli 
That's the GAV that goes into the maven dependencies (dynamically). How should that be referenced instead as it seems that the 3 elements (group, id, version) are all required?

Reply:
I'll try it without the version then and see if everything is happy then.